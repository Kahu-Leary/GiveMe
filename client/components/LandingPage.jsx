import React, { Component } from 'react'
import { Link } from 'react-router-dom'
import { Helmet } from 'react-helmet'

class LandingPage extends React.Component {
  componentDidMount() {
    const backgroundGradient = 'gradient'
    const landing = document.getElementsByClassName('landing')
    if (landing) {
      document.body.classList.add(backgroundGradient)
    }

  }
  render() {
  return (
        <>
        {/* <Helmet >
         <body id='landingBody' />
        </Helmet> */}

            <div className="landing">
              <div className="landingLogo">
                <img className="is-rounded" src='./img/logo-alt.png' />
                <h1>Give Me</h1>
              </div>
              <p>GiveMe is an online platform where kiwis can donate their unwanted items to other kiwis, empowering the NZ community to improve sustainability and reduce the waste caused by unwanted products.</p>

              <div className='landingBtns'>
                <Link className="button browseBtn">Browse</Link>
                <Link className="button registerBtn">Register</Link>
              </div>
            </div>
        </>
  )
 }
}

export default LandingPage
