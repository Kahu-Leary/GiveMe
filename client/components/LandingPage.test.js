import React from 'react'
import { HashRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'

import LandingPage from './LandingPage'

it('renders correctly', () => {
  const tree = renderer
    .create(
      <HashRouter>
        <LandingPage/>
      </HashRouter>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
