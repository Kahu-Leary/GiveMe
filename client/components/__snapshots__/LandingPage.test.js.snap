// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="landing-wrapper"
>
  <div
    className="landingLogo"
  >
    <img
      className="logoImg"
      src="./img/logo-txt-white.png"
    />
  </div>
  <div
    className="landing-text"
  >
    <p>
      GiveMe is an online platform where kiwis can donate their unwanted items to other kiwis, empowering the NZ community to improve sustainability and reduce the waste caused by unwanted products.
    </p>
  </div>
  <div
    className="landingBtnsContainer"
  >
    <a
      className="button browseBtn"
      href="#/home"
      onClick={[Function]}
    >
      Browse
    </a>
    <a
      className="button primaryBtn"
      href="#/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</div>
`;
